providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/common/Util.java

```diff
@@ -0,0 +1,23 @@
+package dev.openfeature.contrib.providers.flagd.resolver.common;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class Util {
+    /**
+     * A helper to block the caller for given conditions.
+     *
+     * @param deadline number of milliseconds to block
+     * @param check    {@link AtomicBoolean} to check for status true
+     */
+    public static void busyWaitAndCheck(final Long deadline, final AtomicBoolean check) throws InterruptedException {
+        long start = System.currentTimeMillis();
+
+        do {
+            if (deadline <= System.currentTimeMillis() - start) {
+                throw new RuntimeException(String.format("Initialization not complete after %d ms", deadline));
```

Comment:
Minor - can change this log line to say something like `Deadline exceeded. Condition did not complete within the deadline`

Reply:
fixed!